src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/EnumStorageShouldBeInt32Tests.cs

```diff
@@ -12,12 +12,214 @@ public class EnumStorageShouldBeInt32Tests : DiagnosticAnalyzerTestBase
     {
         protected override DiagnosticAnalyzer GetBasicDiagnosticAnalyzer()
         {
-            return new BasicEnumStorageShouldBeInt32Analyzer();
+            return new EnumStorageShouldBeInt32Analyzer();
         }
 
         protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
         {
-            return new CSharpEnumStorageShouldBeInt32Analyzer();
+            return new EnumStorageShouldBeInt32Analyzer();
         }
+
+        #region CSharpUnitTests
+
+        [Fact]
+        public void CSharp_CA1028_NoDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+namespace Test
+{
+    public enum TestEnum1 //no violation - because underlying type is Int32
+    {
+        Value1 = 1,
+        Value2 = 2
+    }
+    public static class OuterClass
+    {
+        [Flags]
+        public enum TestEnum2 : long //no violation - because underlying type is Int64 and has Flag attributes
+        {
+            Value1 = 1,
+            Value2 = 2,
+            Value3 = Value1 | Value2
+        }
+        private enum TestEnum3 : byte //no violation - because accessibility is private 
+        {
+            Value1 = 1,
+            Value2 = 2
+        }
+        internal class innerClass
+        {
+            public enum TestEnum4 : long //no violation - because resultant accessibility is private 
+            {
+                Value1 = 1,
+                Value2 = 2
+            }
+        }
+        public static void Main()
+        {
+        }
+    }
```

Comment:
No need for the `Main` method. Yes, it's just test code, but I really feel that every extra line of code that doesn't serve a purpose should be removed. Lots of reasons for that which I can go on and on about :)